File: challenge6.js
require('chromedriver');
var webdriver = require('selenium-webdriver');
var assert = require("chai").assert;
var until = webdriver.until;
var By = webdriver.By;
var Key = webdriver.Key;
var WebElement = webdriver.WebElement;
var fs = require('fs');
//const driverManager = require('../common/driver');
//var screenshot = required('../common/screenshot');
//var filter = require("../common/filters");
//var header = require("../common/headerSearch");
//var results = require("../common/searchResults");

describe("challenge6 suite", function(){
    this.timeout(100000);
    var driver;
    before(async function () {
        // initializing chrome driver
            driver = new webdriver.Builder()
                .withCapabilities(webdriver.Capabilities.chrome())
                .build();
            driver.manage().window().maximize();

            //return driver = await d.getdriver1();
    }

);



    after(function () {
        //return driver.quit();
    });

    it("I open the copart website", function() {
        return driver.get("http://www.copart.com");
    });
    it("It searches for Nissan", async function() {

        var element = await driver.findElement(By.id('input-search'));
        element.sendKeys("Nissan");
        var element2 = await driver.findElement(By.xpath("//*[@id=\"search-form\"]/div/div[2]/button"));
        return await element2.click();

        //lotsearchLotmake
    });
    it("Checks if it user is in search exotic page", async function() {
        var element3 = await driver.getTitle();

        return driver.wait(until.titleIs('Nissan For Auction at Copart - Salvage Cars For Sale'), 4000);

    });
    it("Finds porsche in the list", async function(){

        await driver.wait(until.elementIsVisible(driver.findElement(By.xpath('//table[@id="serverSideDataTable"]//tbody'))));
        var html = await driver.findElement(By.id('serverSideDataTable')).getAttribute('innerHTML');

        return assert.include(html, 'NISSAN');
    });
    it("Picks model button and enters Skyline in the search bar", async function(){
        //*[@id="serverSideDataTable_length"]/label/select
        //
         var element = await driver.wait(until.elementIsVisible(driver.findElement(By.xpath('//*[@id="filters-collapse-1"]/div[1]/ul/li[4]/h4/a[1]/i'))));
         await element.click();
        var element2 = await driver.wait(until.elementIsVisible(driver.findElement(By.xpath('//*[@id="collapseinside4"]/form/div/input'))));
        return await element2.sendKeys('Skyline');



    });

    it("should search the Model filter and enter Skyline", async function(){
        var finalResult; 
        try{
            await driver.wait(until.elementIsVisible(driver.findElement(By.xpath('//*[@id="collapseinside4"]/ul/li'))));
            var html = await driver.findElement(By.id('collapseinside4')).getAttribute('innerHTML');
            console.log(html);
            finalResult = assert.include(html, 'SKYLINE');
            

        }catch(e){
            driver.takeScreenshot().then(function(data){
                fs.writeFileSync('img.png',data, 'base64');
            });
            console.log(e);
            console.log("Did not find SKYLINE");
        }finally{
            return finalResult;
        }


    });



});

File: challenge6.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.common.keys import Keys
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


driver = webdriver.Chrome("../chromedriver.exe")
driver.implicitly_wait(60) 
driver.get("https://www.copart.com/")
driver.maximize_window()
elem = driver.find_element_by_id('input-search')
elem.clear()
elem.send_keys("Nissan")
elem2 =driver.find_element_by_xpath("//*[@id=\"search-form\"]/div/div[2]/button")
elem2.click()
elem3 = driver.title
print(elem3)
if elem3 == 'Nissan For Auction at Copart - Salvage Cars For Sale':
    print("User is in correct website")
else:
    print("User is in correct website")   


elem4 = WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.XPATH,'//*[@id="filters-collapse-1"]/div[1]/ul/li[4]/h4/a[1]/i'))) 
#elem4 = driver.find_element_by_xpath('//*[@id="filters-collapse-1"]/div[1]/ul/li[4]/h4/a[1]/i')
elem4.click()
elem5 = WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.XPATH,'//*[@id="collapseinside4"]/form/div/input'))) 
#elem5 = driver.find_element_by_class('//*[@id="collapseinside4"]/form/div/input')
elem5.send_keys("Skyline")

try:
    WebDriverWait(driver, 10).until(EC.visibility_of_element_located((By.ID,'collapseinside4'))) 
    html = driver.find_element_by_id('collapseinside4').get_attribute('innerHTML')
    print(html)
    if "SKYLINE" in html:
        print("Skyline is present")
    else:
        print("Skyline is not present") 
    
except ValueError:
    driver.save_screenshot("screenshot.png")
    print("Skyline was not found")



